libs/core/src/components/pds-popover/pds-popover.tsx

```diff
@@ -0,0 +1,174 @@
+ * @slot content - HTML content for the popover
+ */
+
+@Component({
+  tag: 'pds-popover',
+  styleUrl: 'pds-popover.scss',
+  shadow: true,
+})
+export class PdsPopover {
+  private contentEl: HTMLElement | null;
+
+  /**
+   * Reference to the Host element
+   */
+  @Element() el: HTMLPdsPopoverElement;
+
+  /**
+   * Determines when the popover is open
+   * @defaultValue false
+   */
+  @State() isOpen = false;
+
+  /**
+   * Id used to reference the component
+   */
+  @Prop() componentId: string;
+
+  /**
+   * Determines whether or not the popover has an arrow
+   * @defaultValue true
+   */
+  @Prop() hasArrow? = false;
+
+  /**
+   * Determines the preferred position of the popover
+   * @defaultValue "right"
+   */
+  @Prop({ reflect: true }) placement:
+    'top'
+    | 'top-start'
+    | 'top-end'
+    | 'right'
+    | 'right-start'
+    | 'right-end'
+    | 'bottom'
+    | 'bottom-start'
+    | 'bottom-end'
+    | 'left'
+    | 'left-start'
+    | 'left-end' = 'right';
+
+  /**
+   * Determines whether or not the popover is visible
+   * @defaultValue false
+   */
+  @Prop({mutable: true, reflect: true}) opened = false;
+
+  /**
+   * Emitted after a popover is closed
+   */
+  @Event() pdsPopoverHide: EventEmitter;
+
+  /**
+   * Emitted after a popover is shown
+   */
+  @Event() pdsPopoverShow: EventEmitter;
+
+  componentDidLoad() {
+    document.addEventListener('click', this.handleGlobalClick);
+  }
+
+  componentDidUpdate() {
+    if (this.opened) {
+      this.showPopover();
+    }
+  }
+
+  componentDidRender() {
+    positionTooltip({elem: this.el, elemPlacement: this.placement, overlay: this.contentEl});
+  }
+
+  /**
+   * Toggles the popover visibility on click
+   */
+  @Method()
+  async togglePopover() {
+    this.opened = !this.opened;
+
+    if (this.opened) {
+      this.handleShow();
+    } else {
+      this.handleHide();
+    }
+  }
+
+  /**
+   * Shows the popover by enabling the opened property
+   */
+  @Method()
+  async showPopover() {
+    this.opened = true;
+  }
+
+  /**
+   * Hides the popover by disabling the opened property
+   */
+  @Method()
+  async hidePopover() {
+    this.opened = false;
+  }
+
+  private handleHide = () => {
+    this.hidePopover();
+    this.pdsPopoverHide.emit();
+  };
+
+  private handleShow = () => {
+    this.showPopover();
+    this.pdsPopoverShow.emit();
+  };
+
+  /**
+   * Closes the popover if the click is not inside the popover
+   */
+  private handleGlobalClick = (event: MouseEvent) => {
+    if(this.opened) {
+      if (!this.el.contains(event.target as Node)) {
+        this.handleHide();
+      }
+    }
+  };
+
+  render() {
+    return (
+      <Host>
+        <div
+          class={`
+            pds-popover
+            pds-popover--${this.placement}
+            ${this.opened ? 'pds-popover--is-open' : ''}
+            ${this.hasArrow ? '' : 'pds-popover--no-arrow'}
+          `}
+        >
+          <span
```

Comment:
Possibly a non-issue, but I see the span element does not resize to match the content.

<img width="264" alt="Screenshot 2023-09-12 at 1 58 08 PM" src="https://github.com/Kajabi/pine/assets/1175111/70f908c0-6da8-4927-8ed0-36bd875c276d">

<img width="269" alt="Screenshot 2023-09-12 at 2 00 00 PM" src="https://github.com/Kajabi/pine/assets/1175111/a0e23980-144a-4982-b37c-3d86071f7360">